index.html

```diff
@@ -288,8 +288,9 @@ <h4 class="panel-title">
           <tbody ng-repeat="bufferline in bufferlines">
             <tr class="bufferline">
               <td class="time">
-                <span class="date" ng-class="::{'repeated-time': bufferline.shortTime==bufferlines[$index-1].shortTime}">
-                  <span class="cof-chat_time cob-chat_time coa-chat_time" ng-bind="::(bufferline.date|date:'HH')"></span><span class="cof-chat_time_delimiters cob-chat_time_delimiters coa-chat_time_delimiters">:</span><span class="cof-chat_time cob-chat_time coa-chat_time" ng-bind="::(bufferline.date|date:'mm')"></span><span class="seconds"><span class="cof-chat_time_delimiters cob-chat_time_delimiters coa-chat_time_delimiters">:</span><span class="cof-chat_time cob-chat_time coa-chat_time" ng-bind="::(bufferline.date|date:'ss')"></span></span>
+                <span ng-if="settings.showtimestamp" class="date" ng-class="::{'repeated-time': bufferline.shortTime==bufferlines[$index-1].shortTime}">
+                  <span ng-if="!settings.showtimestampSeconds" class="cof-chat_time cob-chat_time coa-chat_time" ng-bind-html="::bufferline.formattedTime"></span>
+                  <span ng-if="settings.showtimestampSeconds" class="cof-chat_time cob-chat_time coa-chat_time" ng-bind-html="::bufferline.formattedTimeSeconds"></span>
```

Comment:
These `ng-if`s introduce 2 watchers per line - I suggest just refreshing the buffer when that setting is changed (as a setting change hook) and using the single-binding syntax (`::`) here as well